src/gui/layout/qgslayoutmapwidget.cpp

```diff
@@ -2184,3 +2192,40 @@ void QgsLayoutMapClippingWidget::atlasToggled( bool atlasEnabled )
     mClipToAtlasCheckBox->setChecked( false );
   }
 }
+
+void QgsLayoutMapClippingWidget::invertSelection()
+{
+  for ( int i = 0; i < mLayerModel->rowCount( QModelIndex() ); i++ )
+  {
+    QModelIndex index = mLayerModel->index( i, 0 );
+    Qt::CheckState currentState = Qt::CheckState( mLayerModel->data( index, Qt::CheckStateRole ).toInt() );
+    Qt::CheckState newState = ( currentState == Qt::Checked ) ? Qt::Unchecked : Qt::Checked;
+    mLayerModel->setData( index, newState, Qt::CheckStateRole );
+  }
+}
+
+void QgsLayoutMapClippingWidget::selectAll()
+{
+  for ( int i = 0; i < mLayerModel->rowCount( QModelIndex() ); i++ )
+  {
+    QModelIndex index = mLayerModel->index( i, 0 );
+    mLayerModel->setData( index, Qt::Checked, Qt::CheckStateRole );
```

Comment:
I think here we could use the [checkAll function](https://github.com/qgis/QGIS/blob/ad60cb3a26fe141253e016c2a506e12ab63f78c1/src/core/qgsmaplayermodel.cpp#L77) but while the checkboxes of the layers are all toggled, there is no change on the map clipping side, as if no signal of datachanged was sent. No idea if it's me overlooking something or something missing in the concerned class... Any suggestions?
```suggestion
  mLayerModel->checkAll( Qt::Checked );
```